CMSgov/design-system
# [WNMGDS-1272] Clean up ds-healthcare-gov types

### Changed
- Update the Button overrides
  - Tried to do this without using the old `.d.ts` entryfile, but I failed. As you can see from the commit log, I tried many permutations, but ultimately the problem was that I couldn't figure out how to both export the Button function itself and modify its type in a way that didn't cause TypeScript to yell at me. Fortunately, I think the secondary variation is something we want to get rid of soon anyway
- Use a single `Language` type definition

## How to test
Check out https://github.com/CMSgov/design-system/commit/be7f2fa9231465ed259d21011e3bffcb2c51bbf8, go to `examples/create-react-app-typescript`, `yarn install`, `yarn start`, and maybe `yarn tsc --noEmit`
